src/model/formats/abstract-format.ts

```diff
@@ -0,0 +1,66 @@
+import { ChecklistFile } from '../../../gen/ts/checklist';
+
+declare global {
+  const enum FormatId {}
+}
+
+export type FileExtension = string;
+
+export function getFileExtension(fileName: string): FileExtension {
+  return fileName.slice(fileName.lastIndexOf('.') + 1).toLowerCase();
+}
+
+export abstract class FormatOptions {
+  protected constructor(
+    public readonly description: string,
+    public readonly supportsInput: boolean,
+    public readonly extension?: FileExtension,
+  ) {}
+}
+
+export class DefaultFormatOptions extends FormatOptions {
```

Comment:
Why do we need this additional class, vs making `FormatOptions` non-abstract?

Reply:
Good question! I have reworked `FormatOptions` to more closely match the approach I found for the Dynon format in d0c5bd7. Now I'm much happier with the results, as there is no need for the class hierarchy in the first place, which makes the point moot... how about you?

Reply:
Amended in 2cd517e, unfortunately, casts remove actual parameter checking, so funnier specialization syntax is actually more type-safe.